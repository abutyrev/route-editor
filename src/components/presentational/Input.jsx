import React, { Component } from "react";

class Input extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ""
    };
  }

  changeHandler = e => {
    this.setState({
      value: e.target.value
    });
  };

  submitHandler = e => {
    e.preventDefault();
    this.props.addRoute(this.state.value);
    this.setState({
      value: ""
    });
  };

  render() {
    return (
      <form className="col s12" onSubmit={this.submitHandler}>
        <div className="input-field col s12">
          <input
            type="text"
            id="new_route"
            className="validate"
            value={this.state.value}
            onChange={this.changeHandler}
          />
          <label htmlFor="new_route">Новая точка маршрута</label>
        </div>
      </form>
    );
  }
}

export default Input;
