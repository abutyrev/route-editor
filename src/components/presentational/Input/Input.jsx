import React, { Component } from "react";

class Input extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: "",
      invalidInput: false
    };
  }

  changeHandler = e => {
    this.setState({
      value: e.target.value
    });
  };

  submitHandler = e => {
    e.preventDefault();
    const geocoder = window.ymaps.geocode(this.state.value, {
      results: 1
    });
    geocoder
      .then(res => {
        let firstGeoObject = res.geoObjects.get(0),
          coords = firstGeoObject.geometry.getCoordinates(),
          address = firstGeoObject.getAddressLine();
        this.props.addPoint({ coords, address });
      })
      .catch(err => {
        this.setState({
          invalidInput: true
        });
      });
    this.setState({
      value: ""
    });
  };

  render() {
    return (
      <form className="input-form" onSubmit={this.submitHandler}>
        <div className="input-field">
          <input
            type="text"
            id="new_point"
            className={
              this.state.invalidInput ? "invalid validate" : "validate"
            }
            value={this.state.value}
            onChange={this.changeHandler}
            disabled={!this.props.isMapReady}
          />
          <label htmlFor="new_point">Новая точка маршрута</label>
        </div>
      </form>
    );
  }
}

export default Input;
