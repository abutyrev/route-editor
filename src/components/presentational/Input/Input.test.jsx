import React from "react";
import { shallow, mount } from "enzyme";

import Input from "./Input";

describe("Input component tests", () => {
  const initialState = {
    value: "",
    invalidInput: false
  };
  it("should render correctly", () => {
    const InputComponent = shallow(<Input />);
    expect(InputComponent).toMatchSnapshot();
  });

  it("should render correctly with initial state", () => {
    const InputComponent = shallow(<Input />);
    expect(InputComponent.state()).toEqual(initialState);
  });

  it("check onChange callback", () => {
    const inputComponent = shallow(<Input />);
    inputComponent.find("input").simulate("change", {
      target: {
        value: "aaa"
      }
    });
    expect(inputComponent.state().value).toEqual("aaa");
  });

  /*   it("check onSubmit callback", () => {
    const state = {
      value: 'Петрозаводск',
      invalidInput: false
    },
    addPoint = jest.fn(),
    props = {
      addPoint
    },
    inputComponent = shallow(<Input />);

    inputComponent.setState(state);
    inputComponent.find("form").simulate("submit", {
        preventDefault: () => {}
    });
    expect(inputComponent.submitHandler).toHaveBeenCalledTimes(1);
  }); */

  it("should have error class in error issue", () => {
    const testState = {
        value: "",
        invalidInput: true,
        isMapReady: true
      },
      inputComponent = shallow(<Input />);
    inputComponent.setState(testState);
    expect(inputComponent.find("input").hasClass("invalid")).toEqual(true);
  });

  it("should have disabled attribute while map not ready", () => {
    const testProps = {
        value: "",
        invalidInput: false,
        isMapReady: false
      },
      inputComponent = mount(<Input {...testProps} />);
    expect(inputComponent.find("input").getDOMNode().disabled).toEqual(true);
  });
});
