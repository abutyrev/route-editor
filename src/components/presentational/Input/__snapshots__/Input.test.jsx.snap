// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="input-field col s12"
      >
        <input
          className="validate"
          id="new_point"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="new_point"
        >
          Новая точка маршрута
        </label>
      </div>,
      "className": "col s12",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="validate"
            id="new_point"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <label
            htmlFor="new_point"
          >
            Новая точка маршрута
          </label>,
        ],
        "className": "input-field col s12",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "validate",
            "id": "new_point",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Новая точка маршрута",
            "htmlFor": "new_point",
          },
          "ref": null,
          "rendered": "Новая точка маршрута",
          "type": "label",
        },
      ],
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="input-field col s12"
        >
          <input
            className="validate"
            id="new_point"
            onChange={[Function]}
            type="text"
            value=""
          />
          <label
            htmlFor="new_point"
          >
            Новая точка маршрута
          </label>
        </div>,
        "className": "col s12",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="validate"
              id="new_point"
              onChange={[Function]}
              type="text"
              value=""
            />,
            <label
              htmlFor="new_point"
            >
              Новая точка маршрута
            </label>,
          ],
          "className": "input-field col s12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "validate",
              "id": "new_point",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Новая точка маршрута",
              "htmlFor": "new_point",
            },
            "ref": null,
            "rendered": "Новая точка маршрута",
            "type": "label",
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
